### PR TITLE
align book with cleaned up examples for echo and chat

### PR DESCRIPTION
**Pull request by jhellerstein**

### CODE REVIEW

File: book/example_8_chat_server.md [hunk 1]
```diff
@@ -118,27 +104,27 @@ The `inbound` channel is a source stream that will carry many
...
-pub(crate) async fn run_server(outbound: UdpSink, inbound: UdpStream, graph: Option<GraphType>) {
+pub(crate) async fn run_server(outbound: UdpSink, inbound: UdpStream, opts: Opts) {
     println!("Server live!");
 
     let mut df: Hydroflow = hydroflow_syntax! {
-        // NW channels
+        // Define shared inbound and outbound channels
         outbound_chan = merge() -> dest_sink_serde(outbound);
         inbound_chan = source_stream_serde(inbound)
-            ->  demux(|(msg, addr), tl!(clients, msgs, errs)|
```
Comment (MingweiSamuel) [inline book/example_8_chat_server.md:141]: (Hmm guess I missed that `tl!` before)

File: book/example_8_chat_server.md [hunk 2]
```diff
@@ -315,15 +302,15 @@ when the client receives its sole `ConnectResponse`.
 3. The final pipeline simple pretty-prints the messages received from the server.
 
 ```rust,ignore
-        // send a single connection request on startup
+       // send a single connection request on startup
```
Comment (MingweiSamuel) [inline book/example_8_chat_server.md:305]: missing one space in the indent here

File: book/example_8_chat_server.md [hunk 3]
```diff
@@ -302,8 +289,8 @@ pub(crate) async fn run_client(
 The core logic of the client consists of three dataflow pipelines shown below. 
 
 1. The first pipeline is the "bootstrap" alluded to above.
-It starts with `source_iter` operator on a single, opaque "unit" (`()`) value. This value is available when the client begins, which means 
-this pipeline runs immediately on startup, and generates a single `ConnectRequest` message which is sent to the server.
+It starts with `source_iter` operator that operates over a single, opaque "unit" (`()`) value. This value is available when the client begins, which means
```
Comment (MingweiSamuel) [inline book/example_8_chat_server.md:292]: `/operates/iterates/` ? Or `/operates over/emits/`

File: hydroflow/examples/echoserver/client.rs [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
-    println!("Attempting to connect to server at {:?}", server_addr);
+pub(crate) async fn run_client(outbound: UdpSink, inbound: UdpStream, opts: Opts) {
+    // server_addr is required for client
+    let server_addr = match opts.server_addr {
+        Some(addr) => {
+            println!("Connecting to server at {:?}", addr);
+            addr
+        }
+        None => panic!("Client requires a server address"),
+    };
```
Comment (MingweiSamuel) [inline hydroflow/examples/echoserver/client.rs:16]: This probably ought to be
```rust
let server_addr = opts.server_addr.expect("Client requires a server address");
println!("Connecting to server at {:?}", server_addr);
```

(There might even be a fancy way to handle this logic directly in `clap`)

File: hydroflow/examples/chat/client.rs [hunk 5]
```diff
@@ -28,13 +26,15 @@ fn pretty_print_msg(msg: Message) {
...
-) {
+pub(crate) async fn run_client(outbound: UdpSink, inbound: UdpStream, opts: Opts) {
+    // server_addr is required for client
+    let server_addr = match opts.server_addr {
+        Some(addr) => {
+            println!("Connecting to server at {:?}", addr);
+            addr
+        }
+        None => panic!("Client requires a server address"),
+    };
```
Comment (MingweiSamuel) [inline hydroflow/examples/chat/client.rs:37]: same as above/below
```rust
let server_addr = opts.server_addr.expect("Client requires a server address");
println!("Connecting to server at {:?}", server_addr);
```

**Review by MingweiSamuel — APPROVED** [review]

File: book/example_8_chat_server.md (hunk 2 above)
Comment (jhellerstein) [reply book/example_8_chat_server.md:305]: good eye!

**Pull request merged**